### PR TITLE
fix: fixed editor bugs

### PR DESCRIPTION
**Pull request by qinluhe**
### Fixed Issues
https://github.com/AppFlowy-IO/AppFlowy/issues/4457
https://github.com/AppFlowy-IO/AppFlowy/issues/4456
https://github.com/AppFlowy-IO/AppFlowy/issues/4455
https://github.com/AppFlowy-IO/AppFlowy/issues/4454
https://github.com/AppFlowy-IO/AppFlowy/issues/4453
https://github.com/AppFlowy-IO/AppFlowy/issues/4452
https://github.com/AppFlowy-IO/AppFlowy/issues/4451
https://github.com/AppFlowy-IO/AppFlowy/issues/4450
https://github.com/AppFlowy-IO/AppFlowy/issues/4449
https://github.com/AppFlowy-IO/AppFlowy/issues/4448
https://github.com/AppFlowy-IO/AppFlowy/issues/4447
https://github.com/AppFlowy-IO/AppFlowy/issues/4446

### Feature Preview

---

#### PR Checklist

- [ ] My code adheres to [AppFlowy's Conventions](https://docs.appflowy.io/docs/documentation/software-contributions/conventions)
- [ ] I've listed at least one issue that this PR fixes in the description above.
- [ ] I've added a test(s) to validate changes in this PR, or this PR only contains semantic changes.
- [ ] All existing tests are passing.

### CODE REVIEW

**Comment by appflowy** [comment]
Consider addressing the issues individually, which would make this PR less extensive

File: frontend/appflowy_tauri/src/appflowy_app/components/editor/components/blocks/database/DatabaseList.tsx
Comment (qinluhe) [inline frontend/appflowy_tauri/src/appflowy_app/components/editor/components/blocks/database/DatabaseList.tsx:1]: Added keyboard event: select database list

File: frontend/appflowy_tauri/src/appflowy_app/components/editor/components/blocks/database/Drawer.tsx
Comment (qinluhe) [inline frontend/appflowy_tauri/src/appflowy_app/components/editor/components/blocks/database/Drawer.tsx:1]: Added keyboard events: expand/collapse drawer

File: frontend/appflowy_tauri/src/appflowy_app/components/editor/components/blocks/database/GridBlock.tsx
Comment (qinluhe) [inline frontend/appflowy_tauri/src/appflowy_app/components/editor/components/blocks/database/GridBlock.tsx:1]: Fix grid block style bug and scrolling issues

File: frontend/appflowy_tauri/src/appflowy_app/components/editor/components/blocks/database/GridView.tsx
Comment (qinluhe) [inline frontend/appflowy_tauri/src/appflowy_app/components/editor/components/blocks/database/GridView.tsx:1]: Fix grid block scrolling issue

File: frontend/appflowy_tauri/src/appflowy_app/components/editor/components/blocks/divider/DividerNode.tsx
Comment (qinluhe) [inline frontend/appflowy_tauri/src/appflowy_app/components/editor/components/blocks/divider/DividerNode.tsx:1]: Fix divider style issue

File: frontend/appflowy_tauri/src/appflowy_app/components/editor/components/blocks/math_equation/EditPopover.tsx
Comment (qinluhe) [inline frontend/appflowy_tauri/src/appflowy_app/components/editor/components/blocks/math_equation/EditPopover.tsx:1]: Add keyboard events for formula block

File: frontend/appflowy_tauri/src/appflowy_app/components/editor/components/blocks/text/StartIcon.hooks.tsx
Comment (qinluhe) [inline frontend/appflowy_tauri/src/appflowy_app/components/editor/components/blocks/text/StartIcon.hooks.tsx:1]: Adapt the align function to align the icons of blocks such as checkboxes to the left/right/center along with the text.

File: frontend/appflowy_tauri/src/appflowy_app/components/editor/components/editor/CustomEditable.tsx
Comment (qinluhe) [inline frontend/appflowy_tauri/src/appflowy_app/components/editor/components/editor/CustomEditable.tsx:1]: Disable automatic spelling and completion

File: frontend/appflowy_tauri/src/appflowy_app/components/editor/components/editor/Editor.hooks.ts
Comment (qinluhe) [inline frontend/appflowy_tauri/src/appflowy_app/components/editor/components/editor/Editor.hooks.ts:1]: Fix code block highlighting problem
Fix the problem of left and right cursors in inline block

File: frontend/appflowy_tauri/src/appflowy_app/components/editor/components/editor/Element.hooks.ts
Comment (qinluhe) [inline frontend/appflowy_tauri/src/appflowy_app/components/editor/components/editor/Element.hooks.ts:1]: Fix the problem of inaccurate block selection status

File: frontend/appflowy_tauri/src/appflowy_app/components/editor/components/inline_nodes/inline_formula/FormulaEditPopover.tsx
Comment (qinluhe) [inline frontend/appflowy_tauri/src/appflowy_app/components/editor/components/inline_nodes/inline_formula/FormulaEditPopover.tsx:1]: Formula input box: Add keyboard events

File: frontend/appflowy_tauri/src/appflowy_app/components/editor/components/inline_nodes/inline_formula/InlineFormula.tsx
Comment (qinluhe) [inline frontend/appflowy_tauri/src/appflowy_app/components/editor/components/inline_nodes/inline_formula/InlineFormula.tsx:1]: fix: selection problem of inline formula

File: frontend/appflowy_tauri/src/appflowy_app/components/editor/components/inline_nodes/link/Link.tsx
Comment (qinluhe) [inline frontend/appflowy_tauri/src/appflowy_app/components/editor/components/inline_nodes/link/Link.tsx:1]: fix: selection issues with inline links

File: frontend/appflowy_tauri/src/appflowy_app/components/editor/components/inline_nodes/link/LinkEditContent.tsx
Comment (qinluhe) [inline frontend/appflowy_tauri/src/appflowy_app/components/editor/components/inline_nodes/link/LinkEditContent.tsx:1]: feat: add keyboard event for link actions

File: frontend/appflowy_tauri/src/appflowy_app/components/editor/components/inline_nodes/link/LinkEditInput.tsx
Comment (qinluhe) [inline frontend/appflowy_tauri/src/appflowy_app/components/editor/components/inline_nodes/link/LinkEditInput.tsx:1]: fix: Align flutter’s inline link function

File: frontend/appflowy_tauri/src/appflowy_app/components/editor/components/tools/_shared/ColorPicker.tsx
Comment (qinluhe) [inline frontend/appflowy_tauri/src/appflowy_app/components/editor/components/tools/_shared/ColorPicker.tsx:1]: fix: color picker component bugs and add keyboard event for color picker

File: frontend/appflowy_tauri/src/appflowy_app/components/editor/components/tools/block_actions/AddBlockBelow.tsx
Comment (qinluhe) [inline frontend/appflowy_tauri/src/appflowy_app/components/editor/components/tools/block_actions/AddBlockBelow.tsx:1]: Align flutter's + sign interaction

File: frontend/appflowy_tauri/src/appflowy_app/components/editor/components/tools/block_actions/BlockActions.tsx
Comment (qinluhe) [inline frontend/appflowy_tauri/src/appflowy_app/components/editor/components/tools/block_actions/BlockActions.tsx:1]: Move slash panel to global

File: frontend/appflowy_tauri/src/appflowy_app/components/editor/components/tools/block_actions/BlockActionsToolbar.hooks.ts
Comment (qinluhe) [inline frontend/appflowy_tauri/src/appflowy_app/components/editor/components/tools/block_actions/BlockActionsToolbar.hooks.ts:1]: Fixed the issue where hover on the left toolbar does not appear

File: frontend/appflowy_tauri/src/appflowy_app/components/editor/components/tools/block_actions/BlockActionsToolbar.tsx
Comment (qinluhe) [inline frontend/appflowy_tauri/src/appflowy_app/components/editor/components/tools/block_actions/BlockActionsToolbar.tsx:1]: Added right-click context menu function

File: frontend/appflowy_tauri/src/appflowy_app/components/editor/components/tools/block_actions/BlockOperationMenu.tsx
Comment (qinluhe) [inline frontend/appflowy_tauri/src/appflowy_app/components/editor/components/tools/block_actions/BlockOperationMenu.tsx:1]: fix: add keyboard event for left side menu

File: frontend/appflowy_tauri/src/appflowy_app/components/editor/components/tools/command_panel/Command.hooks.ts
Comment (qinluhe) [inline frontend/appflowy_tauri/src/appflowy_app/components/editor/components/tools/command_panel/Command.hooks.ts:1]: Fix the problem that the slash menu will not appear next time after the @ menu appears.

File: frontend/appflowy_tauri/src/appflowy_app/components/editor/components/tools/command_panel/mention_panel/MentionPanel.hooks.tsx
Comment (qinluhe) [inline frontend/appflowy_tauri/src/appflowy_app/components/editor/components/tools/command_panel/mention_panel/MentionPanel.hooks.tsx:1]: Load mention pages

File: frontend/appflowy_tauri/src/appflowy_app/components/editor/components/tools/command_panel/mention_panel/MentionPanelContent.tsx
Comment (qinluhe) [inline frontend/appflowy_tauri/src/appflowy_app/components/editor/components/tools/command_panel/mention_panel/MentionPanelContent.tsx:1]: feat: add keyboard event for mention panel

File: frontend/appflowy_tauri/src/appflowy_app/components/editor/components/tools/command_panel/slash_command_panel/SlashCommandPanelContent.tsx
Comment (qinluhe) [inline frontend/appflowy_tauri/src/appflowy_app/components/editor/components/tools/command_panel/slash_command_panel/SlashCommandPanelContent.tsx:1]: feat: add keyboard event for slash panel

File: frontend/appflowy_tauri/src/appflowy_app/components/editor/components/tools/selection_toolbar/SelectionActions.tsx
Comment (qinluhe) [inline frontend/appflowy_tauri/src/appflowy_app/components/editor/components/tools/selection_toolbar/SelectionActions.tsx:1]: Fixed an issue with the toolbar appearance mechanism

File: frontend/appflowy_tauri/src/appflowy_app/components/editor/components/tools/selection_toolbar/SelectionToolbar.hooks.ts
Comment (qinluhe) [inline frontend/appflowy_tauri/src/appflowy_app/components/editor/components/tools/selection_toolbar/SelectionToolbar.hooks.ts:1]: Fixed an issue with the toolbar appearance mechanism

File: frontend/appflowy_tauri/src/appflowy_app/components/editor/components/tools/selection_toolbar/SelectionActions.tsx
Comment (qinluhe) [reply frontend/appflowy_tauri/src/appflowy_app/components/editor/components/tools/selection_toolbar/SelectionActions.tsx:1]: Split the buttons used in the toolbar into components

File: frontend/appflowy_tauri/src/appflowy_app/components/editor/plugins/shortcuts/shortcuts.hooks.ts
Comment (qinluhe) [inline frontend/appflowy_tauri/src/appflowy_app/components/editor/plugins/shortcuts/shortcuts.hooks.ts:1]: fix: add shift + backspace and etc key event

File: frontend/appflowy_tauri/src/appflowy_app/components/editor/plugins/withBlockDelete.ts
Comment (qinluhe) [inline frontend/appflowy_tauri/src/appflowy_app/components/editor/plugins/withBlockDelete.ts:1]: Fixed the problem of missing nodes after merging when deleting

Comment (qinluhe) [reply frontend/appflowy_tauri/src/appflowy_app/components/editor/plugins/withBlockDelete.ts:1]: Added processing of selection deletion and deletion behavior including document title

File: frontend/appflowy_tauri/src/appflowy_app/components/editor/plugins/withBlockInsertBreak.ts
Comment (qinluhe) [inline frontend/appflowy_tauri/src/appflowy_app/components/editor/plugins/withBlockInsertBreak.ts:1]: Process the logic that requires adding a new line when the selection is in the divider and carriage return

File: frontend/appflowy_tauri/src/appflowy_app/components/editor/plugins/withBlockMove.ts
Comment (qinluhe) [inline frontend/appflowy_tauri/src/appflowy_app/components/editor/plugins/withBlockMove.ts:1]: Fixed bug when nodes are upgraded to a higher level

File: frontend/appflowy_tauri/src/appflowy_app/components/editor/plugins/withPasted.ts
Comment (qinluhe) [inline frontend/appflowy_tauri/src/appflowy_app/components/editor/plugins/withPasted.ts:1]: Add paste processing use cases

File: frontend/appflowy_tauri/src/appflowy_app/components/editor/provider/utils/action.ts
Comment (qinluhe) [inline frontend/appflowy_tauri/src/appflowy_app/components/editor/provider/utils/action.ts:1]: Fix issue when slate tries to delete text node

File: frontend/appflowy_tauri/src/appflowy_app/components/editor/stores/decorate.ts
Comment (qinluhe) [inline frontend/appflowy_tauri/src/appflowy_app/components/editor/stores/decorate.ts:1]: A draw style that uses a temporary selection to preserve the document's selection when the document loses focus due to the presence of other input components.

File: frontend/appflowy_tauri/src/appflowy_app/components/editor/stores/index.ts
Comment (qinluhe) [inline frontend/appflowy_tauri/src/appflowy_app/components/editor/stores/index.ts:1]: Editor's UI global state

File: frontend/appflowy_tauri/src/appflowy_app/components/editor/stores/inline_node.ts
Comment (qinluhe) [inline frontend/appflowy_tauri/src/appflowy_app/components/editor/stores/inline_node.ts:1]: Pop up the inline block popover

File: frontend/appflowy_tauri/src/appflowy_app/components/editor/stores/selected.ts
Comment (qinluhe) [inline frontend/appflowy_tauri/src/appflowy_app/components/editor/stores/selected.ts:1]: The block selected state

File: frontend/appflowy_tauri/src/appflowy_app/components/editor/stores/slash.ts
Comment (qinluhe) [inline frontend/appflowy_tauri/src/appflowy_app/components/editor/stores/slash.ts:1]: The open state of the slash panel, used by + in the left menu

File: frontend/appflowy_tauri/src/appflowy_app/components/layout/nested_page/NestedPage.hooks.ts
Comment (qinluhe) [inline frontend/appflowy_tauri/src/appflowy_app/components/layout/nested_page/NestedPage.hooks.ts:1]: Solve the lag problem when modifying document title

File: frontend/appflowy_tauri/src/appflowy_app/components/_shared/popover/Popover.hooks.ts
[the remark this answers is not in the document]
Comment (qinluhe) [reply frontend/appflowy_tauri/src/appflowy_app/components/_shared/popover/Popover.hooks.ts:1]: image
image
image

File: frontend/appflowy_tauri/src/appflowy_app/components/database/Database.hooks.ts
[the remark this answers is not in the document]
Comment (qinluhe) [reply frontend/appflowy_tauri/src/appflowy_app/components/database/Database.hooks.ts:1]: Uploading 录屏2024-02-01 18.02.25.mov…

**Pull request merged**